### ISSUE TITLE
[Feature] Time-dependent block implementation

### ISSUE THREAD

**Issue by vytautas-a** [pasqal-io/qadence2-expressions#21]
Current implementation of time-dependent block in `qadence` is using all the usual blocks to build the generator for `HamEvo` with two additional twists. First, the parameter that denotes time is a `Parameter` instance with an additional flag `is_time=True`. Currently it can be created using `TimeParameter` helper function in a similar way to `FeatureParameter` and `VariationalParameter`:
```
def TimeParameter(name: str) -> Parameter:
    """Shorthand for `Parameter(..., trainable=False, is_time=True)`."""
    return Parameter(name, trainable=False, is_time=True)
```

Second, the composite block acting as the generator has a property `is_time_dependent=True` (implemented in `AbstractBlock` class):
```
@property
def is_time_dependent(self) -> bool:
    from qadence.blocks.utils import parameters

    params: list[sympy.Basic] = parameters(self)
    return any(getattr(p, "is_time", False) for p in params)
``` 

Basically, that's all that distinguishes an arbitrary block from a time-dependent one.
